var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -492,9 +501,9 @@ def bjam_python_line(self, spec):
 
         return "using python : {0} : {1} : {2} : {3} ;\n".format(
             spec["python"].version.up_to(2),
-            spec["python"].command.path,
-            spec["python"].headers.directories[0],
-            spec["python"].libs[0],
+            Path(spec["python"].command.path).as_posix(),
+            Path(spec["python"].headers.directories[0]).as_posix(),
+            Path(spec["python"].libs[0]).parent.as_posix(),
```

Comment:
(no change needed) I'll note this is a change to the behavior on linux: before it was taking the full path to the library including `libname.so`, but now provides the directory where the library exists.

The pipelines are succeeding on Linux with builds of boost, and this was originally introduced by https://github.com/spack/spack/commit/ce3ab503ded78d19aae45a938084d1f8670cc625, which focused more on Python than on boost, so I don't think it is a problem.

Reply:
I tested on Mac prior to committing, too, and saw no difference. (even setting it to something crazy like `/tmp` made no difference for me), but was vital on Windows.